plutus-ledger-api/test-plugin/Spec/MintValue/V3.hs

```diff
@@ -0,0 +1,153 @@
+{-# LANGUAGE BlockArguments    #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# OPTIONS_GHC -fexpose-all-unfoldings #-}
+{-# OPTIONS_GHC -fno-full-laziness #-}
+{-# OPTIONS_GHC -fno-ignore-interface-pragmas #-}
+{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
+{-# OPTIONS_GHC -fno-spec-constr #-}
+{-# OPTIONS_GHC -fno-specialise #-}
+{-# OPTIONS_GHC -fno-strictness #-}
+{-# OPTIONS_GHC -fno-unbox-small-strict-fields #-}
+{-# OPTIONS_GHC -fno-unbox-strict-fields #-}
+{-# OPTIONS_GHC -fplugin PlutusTx.Plugin #-}
+
+module Spec.MintValue.V3 where
+
+import PlutusTx.Prelude
+
+import Data.Coerce (coerce)
+import PlutusLedgerApi.Test.V1.Value ()
+import PlutusLedgerApi.Test.V3.MintValue ()
+import PlutusLedgerApi.V1.Value (AssetClass (..), Value (..), flattenValue)
+import PlutusLedgerApi.V3.MintValue (MintValue (..), mintValueBurned, mintValueMinted)
+import PlutusTx.AssocMap qualified as Map
+import PlutusTx.Code (CompiledCode, unsafeApplyCode)
+import PlutusTx.IsData (toBuiltinData)
+import PlutusTx.Lift (liftCodeDef)
+import PlutusTx.List qualified as List
+import PlutusTx.Test.Util.Compiled (cekResultMatchesHaskellValue, compiledCodeToTerm)
+import PlutusTx.TH (compile)
+import Prelude qualified as Haskell
+import Test.QuickCheck qualified as QC
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.QuickCheck (Property, testProperty, (===))
+
+tests :: TestTree
+tests = testGroup "MintValue" [testPropsInHaskell, testPropsInPlinth]
+
+--------------------------------------------------------------------------------
+-- MintValue properties --------------------------------------------------------
+
+prop_MintValueBuiltinData :: Either Value MintValue -> Bool
+prop_MintValueBuiltinData values =
+  let (value, mintValue) =
+        case values of
+          Left v   -> (v, coerce v)
+          Right mv -> (coerce mv, mv)
+   in toBuiltinData mintValue == toBuiltinData value
```

Comment:
Can you convert between Value and MintValue just like that? I thought Value was a particular case of MintValue, meaning some MintValues cannot be converted to Values.

Reply:
I can't think of a `MintValue` that can't be a valid `Value`, e.g. `Value`s can contain negative quantities, just not in tx outputs.

Reply:
The scope of this PR is quite narrowly focused on the `mint` field and addresses the problem with Plinth developers [shooting themselves into the foot](https://github.com/IntersectMBO/plutus/pull/5781) with the `currencySymbolValueOf` function.